lib/libzfs/libzfs_dataset.c

```diff
@@ -2714,9 +2724,7 @@ zfs_prop_get(zfs_handle_t *zhp, zfs_prop_t prop, char *propbuf, size_t proplen,
 		break;
 
 	case ZFS_PROP_ORIGIN:
-		str = getprop_string(zhp, prop, &source);
-		if (str == NULL)
-			return (-1);
+		str = (const char *)&zhp->zfs_dmustats.dds_origin;
```

Comment:
This is probably what's causing the test failures:
```suggestion
		str = zhp->zfs_dmustats.dds_origin;
```

Reply:
The test failure it because `char dds_origin[ZFS_MAX_DATASET_NAME_LEN];` is an empty string, so returns blank, while the test expects this to result in `-`. 

I think what we end up needing is:
```suggestion
		str = (const char *)&zhp->zfs_dmustats.dds_origin;
		if (str != NULL && *str == '\0')
			str = zfs_prop_default_string(prop);
```

Reply:
I think the change I suggested should still be made for consistency (we don't generally take the address of a string and cast it like this).

The case where `str == NULL` is not possible so it would help to remove that check. Otherwise I'm left thinking about "ok but if `str != NULL` is false we then give a NULL pointer to strlcpy?"